kyaml/kio/filters/merge3.go

```diff
@@ -143,27 +172,36 @@ func (ts *tuples) isSameResource(meta1, meta2 yaml.ResourceMeta) bool {
 
 // add adds a node to the list, combining it with an existing matching Resource if found
 func (ts *tuples) add(node *yaml.RNode) error {
-	nodeMeta, err := node.GetMeta()
-	if err != nil {
-		return err
-	}
 	for i := range ts.list {
 		t := ts.list[i]
-		if ts.isSameResource(t.meta, nodeMeta) {
+		if ts.matcher == nil {
```

Comment:
Nit: I think we should just make this check when we create the `tuples` data structure instead of waiting until we are inside the loop.

Reply:
Done.